fix(table): fix that the column width is calculated incorrectly

justwiner opened the pull request:
Fix the bug that the column width is calculated incorrectly when dragging the header

fix #1355

Please make sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer to relative issues for your PR.

Ryan2128 commented:
I think this bug is cause by the `show-overflow-tooltip` of props, but your fix seems to have nothing to do with this prop. Is this fix correct?

This is just my question after the test reproduction, I don’t really understand the logic of it.

justwiner commented:
![image](https://user-images.githubusercontent.com/30335393/106405757-42851780-6472-11eb-91ae-918f58893ef1.png)

justwiner commented:
This is a style bug.  

![image](https://user-images.githubusercontent.com/30335393/106405820-621c4000-6472-11eb-9439-ab01df99f474.png)
I don't think the width here needs to exist, because the width of the cell has been set in `colgroup`, the content inside the cell only needs to be 100%  

![image](https://user-images.githubusercontent.com/30335393/106406017-f8506600-6472-11eb-851b-334a54931fc3.png)

justwiner commented:
haha, there seems to be some problems. I'll have a look later

Commit:
fix(table): fix that the column width is calculated incorrectly

Fix the bug that the column width is calculated incorrectly when dragging the header

fix #1355

justwiner commented:
![image](https://user-images.githubusercontent.com/30335393/106430990-9c540480-64a7-11eb-86f8-51af9265b5b9.png)
At present, I think this is the best solution. The reason for this issue is that the width of the column is calculated incorrectly. If the width of the column is not set, the `realwidth` of the column will be `null` after dragging the header. In fact, the real width is the width attribute.  

But on the other hand, you don't need to set width, it just needs to inherit the width of the parent container.

![image](https://user-images.githubusercontent.com/30335393/106432066-2355ac80-64a9-11eb-95db-7bb88940db20.png)

Ryan2128 commented:
OK
